speechbrain/utils/recipe_tests.py

```diff
@@ -0,0 +1,453 @@
+        will only run tests for ASR recipes.
+    filters: list
+        See above.
+
+    Returns
+    ---------
+    test_script: dict
+        A Dictionary containing recipe IDs as keys and test_scripts as values.
+    test_hparam: dict
+        A dictionary containing recipe IDs as keys and hparams as values.
+    test_flag: dict
+        A dictionary containing recipe IDs as keys and the test flags as values.
+    test_check: dict
+        A dictionary containing recipe IDs as keys and the checks as values.
+    """
+
+    # Dictionary initialization
+    test_script = {}
+    test_hparam = {}
+    test_flag = {}
+    test_check = {}
+
+    # Detect needed information for the recipe tests
+    with open(recipe_csvfile, newline="") as csvf:
+        reader = csv.DictReader(csvf, delimiter=",", skipinitialspace=True)
+        for row in reader:
+            if not (do_test(row, filters_fields, filters, test_field)):
+                continue
+            recipe_id = row[recipe_id_field].strip()
+            test_script[recipe_id] = row[script_field].strip()
+            test_hparam[recipe_id] = row[hparam_field].strip()
+            test_flag[recipe_id] = row[test_field].strip()
+            test_check[recipe_id] = row[check_field].strip()
+
+    return test_script, test_hparam, test_flag, test_check
+
+
+def check_files(
+    check_str, output_folder, recipe_id, pattern=r"file_exists=\[(.*?)\]"
+):
+    """Check if the output folder created by the test has the expected files.
+
+    Arguments
+    ---------
+    check_str: str
+        String summarizing the checks to perform.
+    output_folder: path
+        The path where to check the files.
+    recipe_id: str
+        Unique ID of the recipe.
+    pattern: str
+        The pattern used to extract the list of files to check from check_str.
+
+    Returns
+    ---------
+    check: bool
+        True if all the files are found, False otherwise.
+    """
+
+    check = True
+    files_to_check = re.search(pattern, check_str)
+    files_to_check = files_to_check.group(1).split(",")
+
+    for file_to_check in files_to_check:
+        check_path = os.path.join(output_folder, file_to_check)
+        if not (os.path.exists(check_path)):
+            print(
+                "\tERROR: The recipe %s does not contain the expected file %s"
+                % (recipe_id, check_path)
+            )
+            check = False
+    return check
+
+
+def check_performance(
+    check_str, output_folder, recipe_id, pattern=r"performance_check=\[(.*?)\]"
+):
+    """Checks if the performance achieved by the recipe matches with the
+    expectations. This is done by adding a performance_check entry in the recipe
+    check field of the csv recipe file
+    For instance: performance_check=[train_log.txt, train loss, <=15, epoch: 2]),
+    will check the variable "train_loss" in the train_log.txt at epoch 2. It will
+    raise an error if the train_loss is <=15.
```

Comment:
Should this read "raise an error if the train_loss is > 15" ?
```suggestion
    raise an error if the train_loss is >15.
```